tests/utils/test_det_sal_on_img_dir.py

```diff
@@ -0,0 +1,236 @@
+import numpy as np
+from PIL import Image  # type: ignore
+import json
+from click.testing import CliRunner
+import py
+
+from xaitk_saliency.utils.bin.det_sal_on_img_dir import det_sal_on_img_dir
+
+
+class TestDetSalOnImgDir:
+
+    def test_sal_gen(self, tmpdir: py.path.local) -> None:
+        """
+        Test saliency map generation with RandomDetector, RISEGrid, and
+        DRISEScoring.
+        """
+
+        imgs = [Image.fromarray(np.random.randint(255, size=(100, 100, 3), dtype=np.uint8)) for _ in range(3)]
+
+        img_dir = tmpdir.join('imgs')
+        img_dir.mkdir()
+
+        for i, img in enumerate(imgs):
+            img.save(str(img_dir.join(f"test_{i}.png")))
+
+        config_file = tmpdir.join('conf.txt')
+
+        with config_file.open('w') as outfile:
+            json.dump(TEST_CONFIG, outfile)
+
+        output_dir = tmpdir.join('out')
+
+        runner = CliRunner()
+        runner.invoke(det_sal_on_img_dir, [str(img_dir), str(output_dir), str(config_file)])
+
+        # expected output directories
+        expected_outputs = [output_dir.join(f"test_{i}") for i in range(3)]
+        # sort to normalize order
+        assert sorted(output_dir.listdir()) == sorted(expected_outputs)
+
+    def test_sal_gen_with_overlay(self, tmpdir: py.path.local) -> None:
+        """
+        Test normal functionality with overlaid image option.
+        """
+
+        imgs = [Image.fromarray(np.random.randint(255, size=(100, 100, 3), dtype=np.uint8)) for _ in range(4)]
+
+        img_dir = tmpdir.join('imgs')
+        img_dir.mkdir()
+
+        for i, img in enumerate(imgs):
+            img.save(str(img_dir.join(f"test_{i}.png")))
+
+        config_file = tmpdir.join('conf.txt')
+
+        with config_file.open('w') as outfile:
+            json.dump(TEST_CONFIG, outfile)
+
+        output_dir = tmpdir.join('out')
+
+        runner = CliRunner()
+        runner.invoke(det_sal_on_img_dir, [str(img_dir), str(output_dir), str(config_file), "--overlay-image"])
+
+        # expected output directories
+        expected_outputs = [output_dir.join(f"test_{i}") for i in range(4)]
+        # sort to normalize order
+        assert sorted(output_dir.listdir()) == sorted(expected_outputs)
+
+    def test_config_gen(self, tmpdir: py.path.local) -> None:
+        """
+        Test the generate configuration file option.
+        """
+
+        imgs = [Image.fromarray(np.random.randint(255, size=(100, 100, 3), dtype=np.uint8)) for _ in range(2)]
+
+        img_dir = tmpdir.join('imgs')
+        img_dir.mkdir()
+
+        for i, img in enumerate(imgs):
+            img.save(str(img_dir.join(f"test_{i}.png")))
+
+        config_file = tmpdir.join('conf.txt')
+
+        with config_file.open('w') as outfile:
+            json.dump(TEST_CONFIG, outfile)
+
+        output_dir = tmpdir.join('out')
+
+        output_config = tmpdir.join('config.txt')
```

Comment:
There is a code-smell warning about using this string many times. I am going to flag it as a false-alarm in this case because this is just unit testing.